src/components/index.cjsx

```diff
@@ -47,6 +48,18 @@ SingleTask = React.createClass
     @transferPropsTo(<Task key={id} id={id} />)
 
 
+Practice = React.createClass
+  mixins: [Router.Navigation]
+  render: ->  
+    <BS.Button bsStyle="primary" onClick={@onClick}>Practice</BS.Button>
+  onClick: ->
+    CourseActions.loadPractice(courseId)
+    CourseStore.on 'change', ()=>
```

Comment:
Hmm, interesting! I remember having to jump through several hoops to implement `.create` and `.created` but it looks like a more general solution (more Promise'ish, like this) should be adopted instead.

Did you try using something like `CourseActions.loadedPractice.on 'trigger', =>`  to make it clearer which event you're interested in?